### DISCUSSION TITLE
GIRIER Tuya Wifi Smart Dimmer Switch Module 10A

### DISCUSSION THREAD

**Discussion by martin072** [arendst/Tasmota#14506] [Q&A] [answered]
I just got a GIRIER Tuya Wifi Smart Dimmer Switch Module 10A from AliExpress, however, the chip doesn't look familiar to me at all.. Its branded as "16ew/8.24" with a pcb board with version number ZH-Z06 REV:A 

Anyone any idea if this is an ESP compatible chip? of able to be made compatible?

see pictures below
![IMG_2321](https://user-images.githubusercontent.com/24524506/150018786-1863ff50-2c15-48f3-b01f-62d0be7845e0.png)
![IMG_2322](https://user-images.githubusercontent.com/24524506/150018805-723274fc-06e1-4731-93fa-78ab4147c3db.png)

**Comment by sfromis** [comment, marked as answer]
The form factor with 7/8/7 pins on the edge does not look like one of the usual ESP8266 based modules. Maybe there are clues on the back side of the small blue board, or on the back side of the smallest green PCB. If you can use a Wifi tool to get the MAC address (like in pairing mode), you can look it up on a site like macvendors.com to check if it is Espressif.

With 8 pins to the main (larger green) PCB, it should be feasible to work out what those pins are, and connect an ESP module instead, using wires instead of worrying about matching pinout.

**Reply by martin072** [reply]
Haven't powered it on yet, I did de-solder the little pcb, and on the lower side is only a HK HK32F030MF4P6, which is an ARM MCU, probably the MCU used by tuya for the serial communication. Most likely it will be an ARM wifi chip I Guess..

I am hoping this device actually uses a simple PWM signal (perhaps with a ZC input). so I can change that board with a small ESP

**Reply by sfromis** [reply]
My guess would be that the MCU used for wifi is inside the metal cover, with the ARM chip you found acting as the main processor, "TuyaMCU", using the wifi MCU as a "wifi modem". That structure is very common among Tuya devices.

And as PWM is quite common for controlling the high power side of dimmers, the odds should be good of a feasible Frankenstein job. Doing a bit of espionage with the factory firmware still in place can also be quite illustrative....

**Reply by martin072** [reply]
So I tried to probe out the pins that run to the main board, and found the obvious ones like 3.3v, GND, RST. Also found a pin that leads to the input of an  optocoupler, so I will assume this takes a pwm. 

I have two other pins that I cannot make sense of. It might look like a zero-cross input, but there are two. See attached drawing and small probe from an oscilloscope. Q3 and Q4 run to the small mainboard, assuming they are inputs off some sort?

Maybe some electronic brains here that could try to explain what these are for?


![IMG_0415](https://user-images.githubusercontent.com/24524506/150429071-613bb417-435a-49fe-bd88-f15acd6db6c8.JPG)
![IMG_0414](https://user-images.githubusercontent.com/24524506/150429079-d15a9801-cbcb-46b7-bc4b-8c1dfd2065b4.JPG)
![IMG_0413](https://user-images.githubusercontent.com/24524506/150429083-654cdb4c-bb25-47c2-9ec5-1977f5b8ea02.JPG)
![IMG_0412](https://user-images.githubusercontent.com/24524506/150429086-f7c2c6b3-3f28-466b-ac56-5f1e29faacbb.JPG)
![IMG_2328](https://user-images.githubusercontent.com/24524506/150429089-965e33e3-3411-46d1-8137-8743414bd520.png)

**Comment by MaWork77** [comment]
![CamScanner 03-11-2022 16 20-Pinout](https://user-images.githubusercontent.com/57617605/158000015-0cf44b1e-4fa6-44d8-b84f-5758e1601372.jpg)
![CamScanner 03-11-2022 16 20_2](https://user-images.githubusercontent.com/57617605/158000034-9fa63524-fd25-44e8-811c-26920e999f23.jpg)
![CamScanner 03-11-2022 16 20_3](https://user-images.githubusercontent.com/57617605/158000037-c12f07ae-91c1-4210-b3cf-8a9116dc06f0.jpg)
![CamScanner 03-11-2022 16 20_4](https://user-images.githubusercontent.com/57617605/158000038-b723266f-4114-4c63-9c6f-69531fd26fcc.jpg)

**Reply by sfromis** [reply]
Did you check the MAC? Easy way the check for an ESP chip inside. 

To me, the pinout visible on the back side of the small blue PCB does **not** look like an ESP chip inside. Unless the silk screen print does not match, looks like you're out of luck.

**Reply by kalayda** [reply]
Where the bottom 10 holes are connected? Can it be used to flash BK7231T?

**Comment by MaWork77** [comment]
![yjr40qlu4k581 (Mittel)](https://user-images.githubusercontent.com/57617605/158014757-7b3c28e3-061f-4d60-8f2b-64d5801b4e18.jpg)

it's a Beken BK7231 Chip. I install in my module a LM1 module with a ESP Chip. 
I'm still testing, but switching on by script is already possible

**Comment by btsimonh** [comment]
If you want to use the module without modification, search 'OpenBeken' - we have the start of an alternative firmware for the bk7231 based devices.  It's not TAS, but it does work.  Forum discussion [here](https://www.elektroda.com/rtvforum/viewtopic.php?p=19934898#19934898).
I have modified two direct PWM devices so far, a RGBWW and RGB.
TuyaMCU is in development - serial logs welcome.

**Comment by rtrabold** [comment]
Hi, I have the same dimmer but is REV:B, is the same but it came with a CB3S module. I sucessfully flashed OpenBeken on it.
The issue now is to find correct TuyaMCU configuration, dpIds and function. Any clue about it?
Tested with:
```
startDriver TuyaMCU
setChannelType 1 toggle
setChannelType 2 dimmer
tuyaMcu_setDimmerRange 0 1000
linkTuyaMCUOutputToChannel 1 1 1
linkTuyaMCUOutputToChannel 2 2 2
```

But nothing happens....

**Reply by sfromis** [reply]
The Tasmota project is not a great place to ask about how to use OpenBeken....

**Reply by kalayda** [reply]
@rtrabold Could you please explain how did you connect board to flash it?
I have Aubess dimmer with ZH-Z06-A REV: A. Suspect it has BK7231T onboard.  For using ESPHome going to flash libertuya with help of BKWriter app.
But can't understand, which pins to connect with USB-UART connector. Should I use some of ten holes on the board?

![photo_2022-10-27_21-26-22](https://user-images.githubusercontent.com/2320537/198370664-cec31d36-2db8-4d63-9890-381634588fc2.jpg)
![photo_2022-10-27_21-26-22 (2)](https://user-images.githubusercontent.com/2320537/198370670-b4963d5f-da65-44c8-a8c5-1e982390fa22.jpg)
![photo_2022-10-27_21-26-22 (3)](https://user-images.githubusercontent.com/2320537/198370672-90e7a437-e517-4528-aed3-4fdbc98f0b79.jpg)

**Comment by btsimonh** [comment]
@kalayda - it's not very polite to ask the questions here about BK devices.  
Search here:
https://www.elektroda.pl/rtvforum/find.php?q=openbeken
or here:
https://github.com/openshwprojects/OpenBK7231T_App
br,
Simon

**Comment by openshwprojects** [comment]
This device is not using TuyaMCU. It has a custom protocol.
Details:
https://www.elektroda.com/rtvforum/topic3985978.html

**Comment by kroon040** [comment]
I have a similiair device. It has the cb3s form tuya. I try to replace the module with the WT0132C6-S5 (https://manuals.plus/gl/wireless-tag/wt0132c6-s5-wt0132c6-s5u-wi-fi-6-module-manual?), and esp32c6 pin compatible as the esp12.

I have already flash it with tasmota , but I need to solder it. I also need to figure out, if it could work with tasmota.  also need to find out the gpio's for the esp32 to let it work.
